api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -639,5 +647,12 @@ Object writeReplace() throws ObjectStreamException {
     protected Type.TypeID typeId() {
       return Type.TypeID.BINARY;
     }
+
+    @Override
+    public String toString() {
+      byte[] binary = new byte[value().remaining()];
+      value().duplicate().get(binary);
+      return "0x" + BaseEncoding.base16().encode(binary);
```

Comment:
I'm not sure if we have any other places where we convert a `ByteBufer` to a string.

If there are any, it seems like potentially we should add the stringify function to the `ByteBuffer` utility class and then try to use it uniformly everywhere. You don't necessarily need to update the other places to use them in this PR, but would you mind taking a look and seeing if you find any?

Reply:
Follow up question: Are there cases where people might not use base16 for the representation or where the leading zero might not be used? I don't think so for Spark, but I'm less sure with Flink for example.

For Spark, looking at the documentation, it seems like it's just `X`, without the leading zero: https://spark.apache.org/docs/latest/sql-ref-literals.html#binary-literal

I'm not as sure about Trino, Flink, and other systems though.

For Trino, it looks to also start with the capital `X` (without the zero): https://trino.io/docs/current/language/types.html#varbinary

Looking it up, for Flink, it looks like maybe it's actually represented as `BINARY(n)` or `VARBINARY(n)`. This is the one I'm least sure about: https://nightlies.apache.org/flink/flink-docs-release-1.13/docs/dev/table/types/#binary

Reply:
This should just be for debugging. If we are converting a literal to a string and back, then that's a problem.

Reply:
This should just be for debugging.

In the base class, the toString method is `String.valueOf(value)` (In StringLiteral, it is `"\"" + value() + "\""`).

For other types of literals, the return value of toString is clear. But for ByteBuffer, the return value is `java.nio.HeapByteBuffer[pos=0 lim=16 cap=16]`.

So I implemented the literal toString method of  ByteBuffer type, for logging.